beanstalk_worker/worker_start_scan.py

```diff
@@ -26,212 +26,340 @@
+        for i in list_logs:
+            if i == "" or i.startswith("Repodata"):
+                continue
+            i_split = i.split(" ", 1)
+            package_list.append(i_split[0])
+
+        return package_list
+
+    def mount_image(self, image_under_test):
+        """
+        Mount image on local file system
+        """
+        # get the SHA ID of the image.
+        image_id = self.atomic_object.get_input_id(image_under_test)
+        image_rootfs_path = os.path.join("/", image_id)
 
-    logger.log(level=logging.INFO,
-               msg="Mounting rootfs %s on %s" % (image_id, image_rootfs_path))
+        # create a directory /<image_id> where we'll mount image's rootfs
+        os.makedirs(image_rootfs_path)
 
-    mount_object.mount()
+        # configure options before mounting the image rootfs
+        logger.log(level=logging.INFO,
+                   msg="Setting up system to mount image's rootfs")
+        self.mount_object.mountpoint = image_rootfs_path
+        self.mount_object.image = image_id
+        # mount the rootfs in read-write mode. else yum will fail
+        self.mount_object.options = ["rw"]
 
-    logger.log(level=logging.INFO,
-               msg="Successfully mounted image's rootfs")
+        logger.log(
+            level=logging.INFO,
+            msg="Mounting rootfs %s on %s" % (image_id, image_rootfs_path))
 
-    cmd = "atomic scan --scanner=%s --rootfs=%s %s" % \
-        ("pipeline-scanner", image_rootfs_path, image_id)
+        self.mount_object.mount()
 
-    logger.log(level=logging.INFO,
-               msg="Executing atomic scan: %s" % cmd)
+        logger.log(level=logging.INFO,
+                   msg="Successfully mounted image's rootfs")
+        return image_rootfs_path
 
-    process = subprocess.Popen([
-        'atomic',
-        'scan',
-        "--scanner=pipeline-scanner",
-        "--rootfs=%s" % image_rootfs_path,
-        "%s" % image_id
-    ], stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+    def unmount_image(self, mount_object):
+        """
+        Unmount image using the Atomic mount object
+        """
+        logger.log(
+            level=logging.INFO,
+            msg="Unmounting image's rootfs from %s" % mount_object.mountpoint)
+        # TODO: Error handling and logging
+        mount_object.unmount()
 
-    out, err = process.communicate()
+    def remove_image(self, image_rootfs_path, image_under_test):
+        """
+        Removes the mounted image rootfs path as well as
+        removes the image using docker
+        """
+        os.rmdir(image_rootfs_path)
+        logger.log(
+            level=logging.INFO,
+            msg="Removing the image %s" % image_under_test)
+        conn.remove_image(image=image_under_test, force=True)
+
+    def run(self, image_under_test):
+        """
+        Run the scanner
+        """
+        self.mount_image(image_under_test)
+        image_id = self.atomic_object.get_input_id(image_under_test)
```

Comment:
`image_id` is being fetched in `mount_image` function as well. Maybe return it from that function to avoid repetition or maybe initialize `self.image_id` in `__init__` function with the value `None` and set it once when `run` method is executed?